en/lessons/ecto/changesets.md

```diff
@@ -0,0 +1,372 @@
+title: Changesets
+---
+
+In order to insert, update or delete data from the database, `Ecto.Repo.insert/2`, `update/2` and `delete/2` require a changeset as their first parameter.
+But what exactly are changesets?
+
+A familiar task for almost every developer is checking input data for potential errors — we want to make sure that data is in the right state, before we attempt to use it for our purposes.
+
+Ecto provides a complete solution for working with data changes in the form of the `Changeset` module and data structure.
+In this lesson we're going to explore this functionality and learn how to verify data's integrity, before we persist it to the database.
+
+{% include toc.html %}
+
+## Creating your first changeset
+
+Let's look at an empty `%Changeset{}` struct:
+
+```elixir
+iex> %Ecto.Changeset{}
+#Ecto.Changeset<action: nil, changes: %{}, errors: [], data: nil, valid?: false>
+```
+
+As you can see, it has some potentially useful fields, but they are all empty.
+
+For a changeset to be truly useful, when we create it, we need to provide a blueprint of what the data is like.
+What better blueprint for our data than the schemas we've created the define our fields and types?
+
+To save us some time, let's re-use the schema we created in the previous lesson:
+
+```elixir
+defmodule Example.Person do
+  use Ecto.Schema
+
+  schema "people" do
+    field :name, :string
+    field :age, :integer, default: 0
+  end
+end
+```
+
+### `cast/4`
+
+To create a changeset using the `Example.Person` schema, we are going to use [`Ecto.Changeset.cast/4`](https://hexdocs.pm/ecto/Ecto.Changeset.html#cast/4).
+The first parameter should be our starting data which for us will be an empty `%Person{}` but could be an existing record.
+The secord parameter `cast/4` expects is the data changes as a map.
+Our third parameter is a list of fields we're accepting values for, fields not included in this list will be ignored.
+The foruth and final parameter are changeset options which we'll explore later, for now we'll focus on the first three:
+
+```elixir
+iex> alias Example.Person
+iex> Ecto.Changeset.cast(%Person{}, %{"name" => "Jack"}, [:age, :name])
+#Ecto.Changeset<
+ action: nil,
+ changes: %{name: "Jack"},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+ ```
+
+We can see that we've created a new changeset and this time it's populated with values, most importantly: `:changes` and `:valid?`.
+
+Before we go too much further let's quickly take a look at how permitted fields work:
+
+ ```elixir
+iex> Ecto.Changeset.cast(%Person{}, %{"email" => "jack@example.com", "name" => "Jack"}, [:name])
+#Ecto.Changeset<
+ action: nil,
+ changes: %{name: "Jack"},
+ errors: [],
+ data: #Person<>,
+ valid?: true
+>
+```
+
+What we see is the same valid changeset from before with no mention of the `"email"` value because it was not explicitly permitted.
+
+### `change/2`
+
+If the source of the data change is truthworthy `change/2` can be a viable alternative to `cast/4`.
```

Comment:
`trustworthy,`
Am I just misguided on the use of comma or is it really needed here?